src/server/internalServer.cpp

```diff
@@ -505,10 +548,10 @@ std::unique_ptr<Response> InternalServer::handle_suggest(const RequestContext& r
   }
 
   /* Propose the fulltext search if possible */
-  if (reader->hasFulltextIndex()) {
+  if (archive->hasFulltextIndex()) {
```

Comment:
Aah right! It can be a nullptr. This block should be executed only if we are not dealing with a nullptr, and if it is, we should be returning a 404(like we do in handle_search). Perhaps we should check if the archive is a nullptr and return a 404 in such a case before we actually start handling the search.

Reply:
Yes, test and early return is good here.